libs/order-list/src/lib/components/cancel-order-dialog/cancel-dialog.tsx

```diff
@@ -0,0 +1,65 @@
+import { OrderStatus } from '@vegaprotocol/types';
+import { Dialog, Intent } from '@vegaprotocol/ui-toolkit';
+import type { Order, VegaTxState } from '@vegaprotocol/wallet';
+import { VegaTxStatus, VegaOrderTransactionDialog } from '@vegaprotocol/wallet';
+import { useEffect } from 'react';
+
+interface CancelDialogProps {
+  orderDialogOpen: boolean;
+  setOrderDialogOpen: (isOpen: boolean) => void;
+  finalizedOrder: Order | null;
+  transaction: VegaTxState;
+  reset: () => void;
+}
+
+export const CancelDialog = ({
+  orderDialogOpen,
+  setOrderDialogOpen,
+  finalizedOrder,
+  transaction,
+  reset,
+}: CancelDialogProps) => {
+  const getDialogIntent = () => {
+    if (finalizedOrder) {
```

Comment:
**nitpick (non-blocking)**: I found this if logic a little hard to parse because `finalizedOrder` implied to me that we were *placing* an Order, whereas obviously we're waiting for the *updated* order to come in and to confirm that it's actually cancelled.

It's not worth making a change for this.

Reply:
updatedOrder might be better